### PR TITLE
Add more tests for static profile

### PR DESCRIPTION
**Pull request by BrendanBurkhart**
Add more tests for static profile edge cases, handle more edge cases. Clean up static profile chunk generation flow a little bit.

### CODE REVIEW

File: src/main/java/frc/robot/motion/StaticProfile.java [hunk 1]
```diff
@@ -46,65 +46,110 @@ public StaticProfile(double currentVelocity, double currentPosition, double targ
...
-            chunk = Chunk.createVelocityTransition(startVelocity, 0, maxAccel, maxDecel);
+    private ArrayList<Chunk> computeChunks(ArrayList<Chunk> chunks, double currentVelocity, double remainingDistance) {
+        final Chunk chunk;
+
+        double stoppingDistance = 0.5 * (Math.abs(currentVelocity) / maxDecel) * currentVelocity;
+        double targetDirection = Math.signum(remainingDistance);
+        double currentDirection = Math.signum(currentVelocity);
+
+        /***
+         * If going in the wrong direction and at start of profile: transition to
```
[the remark this answers is not in the document]
Comment (calebeby) [reply src/main/java/frc/robot/motion/StaticProfile.java]: Yeah, I think so

File: src/main/java/frc/robot/motion/StaticProfile.java [hunk 2]
```diff
@@ -46,65 +46,91 @@ public StaticProfile(double currentVelocity, double currentPosition, double targ
...
+        // Otherwise, must be going at max speed
+        else {
+            // If there is excess time to stop
+            if (Math.abs(stoppingDistance) < Math.abs(remainingDistance)) {
+                // max velocity transition - continue at max speed for efficiency
+                chunk = Chunk.createConstantVelocity(maxVelocity * targetDirection,
+                        remainingDistance - stoppingDistance);
+            }
+            // Else if stopping distance == remaining distance
+            else if (Math.abs(stoppingDistance - remainingDistance) < epsilon) {
```
Comment (calebeby) [inline src/main/java/frc/robot/motion/StaticProfile.java]: Can we remove the epsilon here?
